### PR TITLE
feat: Safe creation AB test

### LINKED ISSUE #968: [Safe Creation] A/B test the new design
## What is the feature about
We need to analyze how the new design performs against the old one.

The most correct way to do it is to run both versions in parallel and measure the drop-off rate in both.

## The list of requirements

* The new Safe creation should be on a new URL (e.g. `/new-safe/create`) and its components in a corresponding new folder
* The old Safe creation should stay on the old URL
* Add a new A/B hook that would flip a coin and save the variant to the localStorage
* If a user with the B coin goes to the Safe creation, redirect them to `/new-safe/create`
* Add the variant from the localStorage to all GTM events

## The A/B hook
The hook should be generic and work for other A/B tets in the future.

So it should be something like this:

```
const useABTest = (name: string): boolean => {
  const [isB, setB] = useLocalStorage<boolean>(`AB__${name}`, false)
  
  useEffect(() => {
    // Flip the coin if the variant hasn't been determined yet
    setB((prev) => prev ?? Math.random() > 0.5)
  }, [setB])
  
  return isB
}
```

## Links
Epic: https://github.com/safe-global/web-core/issues/864

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #968

## How this PR fixes it

An "coin toss" decides which Safe creation flow to use (new or old) and stores this flag locally. The chosen flow is also sent with every analytics event.

## How to test it

Open Safe creation (from `/welcome`) and note which flow is chosen. Return to `/welcome`, delete the `SAFE_v2__AB__newSafe` `localStorage` entry and refresh the page. Creating a Safe again may choose a different flow. Repeat until the other is taken.

## Analytics changes

`{ abTest?: string }` is sent with events in Safe load/creation, labelled "New" or "Old" accordingly.

### CODE REVIEW

File: src/config/routes.ts [hunk 1]
```diff
@@ -5,13 +5,16 @@ export const AppRoutes = {
...
-  createSafe: '/create-safe',
   apps: '/apps',
   addressBook: '/address-book',
   balances: {
     nfts: '/balances/nfts',
     index: '/balances',
   },
+  newSafe: {
+    create: '/new-safe/create',
+    add: '/new-safe/add',
```
Comment (iamacook) [inline src/config/routes.ts:16]: This was added in anticipation of the add Safe revamp.

File: src/hooks/useNewSafeRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { AppRoutes } from '@/config/routes'
+
+export const NEW_SAFE_AB_TEST_NAME = 'newSafe'
+
+const useNewSafeRoutes = () => {
+  const shouldUseNewRoute = useABTest(NEW_SAFE_AB_TEST_NAME)
+
+  return {
+    createSafe: shouldUseNewRoute ? AppRoutes.newSafe.create : AppRoutes.open,
+    addSafe: shouldUseNewRoute ? AppRoutes.newSafe.add : AppRoutes.load,
```
Comment (iamacook) [inline src/hooks/useNewSafeRoutes.ts:11]: `AppRoutes.newSafe.add` uses the `AppRoutes.load` component for now.

**Comment by katspaugh** [comment]
It would be much simpler to just add a conditional redirect on the old URL.

**Review by katspaugh — CHANGES_REQUESTED** [review]
To make my point about the redirect clear, instead of the `useNewSafeRoutes` spread across many components, the A/B test should have just one entry point: the old Safe creation route.
On that page, do a simple URL redirect to the alternative URL if the user is in B.

As a side note, I would split this PR into two:
* A/B test system implementation
* The Safe creation test

**Comment by iamacook** [comment]
> As a side note, I would split this PR into two:
> 
>     * A/B test system implementation
> 
>     * The Safe creation test

I'll work on this now.

**Pull request closed without merging**